float64 px_coord
float64 py_coord
float64 delta_px
float64 depth
